misc/dbt-materialize/dbt/include/materialize/macros/deploy/deploy_get_objects.sql

```diff
@@ -0,0 +1,60 @@
+-- Copyright Materialize, Inc. and contributors. All rights reserved.
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License in the LICENSE file at the
+-- root of this repository, or online at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+{% macro deploy_get_objects(debug=False) %}
+    {% set clusters = {} %}
+    {% set schemas = {} %}
+
+    {# Get ignored clusters and schemas from project variables #}
+    {% set deployment_vars = var('deployment', {}).get(target.name, {}) %}
+    {% set ignored_clusters = deployment_vars.get('ignored_clusters', []) %}
+    {% set ignored_schemas = deployment_vars.get('ignored_schemas', []) %}
+
+    {% if debug %}
+        {{ log("Debug: Starting deploy_get_objects macro", info=True) }}
+        {{ log("Debug: Ignored clusters: " ~ ignored_clusters, info=True) }}
+        {{ log("Debug: Ignored schemas: " ~ ignored_schemas, info=True) }}
+    {% endif %}
+
+    {# Add cluster and schema from the current target #}
+    {% if target.cluster and target.cluster not in ignored_clusters %}
+        {% do clusters.update({target.cluster: true}) %}
+    {% endif %}
+    {% if target.schema and target.schema not in ignored_schemas %}
+        {% do schemas.update({target.schema: true}) %}
+    {% endif %}
+
+    {# Add clusters and schemas from the models #}
+    {% for node in graph.nodes.values() %}
+        {% if node.resource_type == 'model' %}
```

Comment:
Tests and seeds can also be associated with clusters; do we need to account for those?